### PR TITLE
Use Enum to enforce stricter types (⚠ php 8.1)

### PR DESCRIPTION
**Pull request by ildyria**
# Breaking change for php8.0

- Add Enum for SizeVariants
- Add Enum for Columns types (sorting)
- Add Enum for Order (sorting)
- Add Enum for DownloadType (basically SizeVariants + LivePhoto)
- Add Enum for Severity (logging)
- Add Enum for Update status (migration required, up to date, need update)
- Renames FULL to ORIGINAL

Due to the breaking change of php version, suggested Milestone 4.7.0

### CODE REVIEW

**Comment by ildyria** [comment]
Requires https://github.com/LycheeOrg/Lychee-front/pull/361

File: README.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ Lychee is a free photo-management tool, which runs on your server or web-space.
 
 ## Installation
 
-To run Lychee, everything you need is a web-server with PHP 8.0 or later and a MySQL-Database. Follow the instructions to install Lychee on your server. This version of Lychee is built on the Laravel framework. To install:
+To run Lychee, everything you need is a web-server with PHP 8.1 or later and a MySQL-Database. Follow the instructions to install Lychee on your server. This version of Lychee is built on the Laravel framework. To install:
```
Comment (qwerty287) [inline README.md]:
```suggestion
To run Lychee, everything you need is a web-server with PHP 8.1 or later and a database (MySQL/MariaDB, PostgreSQL or SQLite). Follow the instructions to install Lychee on your server. This version of Lychee is built on the Laravel framework. To install:
```

File: app/Enum/DonwloadVariantType.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace App\Enum;
+
+/**
+ * Enum SizeVariantType.
+ */
+enum DonwloadVariantType: string
```
Comment (qwerty287) [inline app/Enum/DonwloadVariantType.php]:
```suggestion
/**
 * Enum DownloadVariantType.
 */
enum DownloadVariantType: string
```

Please also check the other comments. Most are not correct

Comment (ildyria) [reply app/Enum/DonwloadVariantType.php]: done. [5a864a8](https://github.com/LycheeOrg/Lychee/pull/1618/commits/5a864a8e7e2db244bc2f007cbe3462710ce8a12f)

File: README.md (hunk 1 above)
Comment (ildyria) [reply README.md]: Done [5a864a8](https://github.com/LycheeOrg/Lychee/pull/1618/commits/5a864a8e7e2db244bc2f007cbe3462710ce8a12f).

File: app/Enum/DonwloadVariantType.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+namespace App\Enum;
+
+/**
+ * Enum DonwloadVariantType.
+ *
+ * Because there are Live photos, it is not possible to merge those types with SizeVariantType
+ * This essencially accomplish the sames and provides a type mapping to ensure correct conversions.
+ */
+enum DonwloadVariantType: string
```
Comment (qwerty287) [inline app/Enum/DonwloadVariantType.php]:
```suggestion
enum DownloadVariantType: string
```

Comment (ildyria) [reply app/Enum/DonwloadVariantType.php]: Done. [66a9703](https://github.com/LycheeOrg/Lychee/pull/1618/commits/66a97033b3ea559f13f315e34ffaf8107473b84b)

**Review by qwerty287 — COMMENTED** [review]
Just storing viewed files. Code wise looks good, but I'd like to test after 4.6.4 is out

**Review by qwerty287 — APPROVED** [review]

**Pull request merged**